server/src/write_buffer.rs

```diff
@@ -34,16 +33,11 @@ impl WriteBufferConsumer {
     pub fn new(
         mut write_buffer: Box<dyn WriteBufferReading>,
         db: Arc<Db>,
-        registry: &MetricRegistry,
+        registry: &metric::Registry,
     ) -> Self {
         let shutdown = CancellationToken::new();
 
-        let ingest_domain = registry.register_domain_with_attributes(
-            "write_buffer",
-            vec![KeyValue::new("db_name", db.rules().name.to_string())],
-        );
-
-        let ingest_metrics = metrics::WriteBufferIngestMetrics::new(Arc::new(ingest_domain));
+        let ingest_metrics = WriteBufferIngestMetrics::new(registry, db.rules().name.to_string());
```

Comment:
```suggestion
        let ingest_metrics = WriteBufferIngestMetrics::new(registry, &db.rules().name);
```

probably doesn't matter but I think the argument is `impl Into<String>`